### PR TITLE
[IMP] mail, *: auto add implicit attributes of identifying fields

### PR DESCRIPTION
**Pull request by oomsveta**
Automatically add `required` and `readonly` on identifying fields and remove them from field definition.

Task-2955910.
\* = calendar, hr, im_livechat, rating, snailmail, website_livechat

Enterprise: odoo/enterprise#30511

### CODE REVIEW

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/model/model_manager.js [hunk 1]
```diff
@@ -1266,6 +1272,10 @@ export class ModelManager {
             for (const [fieldName, field] of Object.entries(model.__fieldMap)) {
                 if (field.identifying) {
                     model.__identifyingFieldNames.add(fieldName);
+                    model.__fieldMap[fieldName].readonly = true;
```
Comment (seb-odoo) [inline addons/mail/static/src/model/model_manager.js:1275]: should be in field constructor

**Comment by oomsveta** [comment]
@robodoo r-

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**